test/service/test_response.py

```diff
@@ -1476,6 +1479,149 @@ def test_disease_facet(self):
+                    },
+                    {
+                        'term': {
+                            'value': '55-59',
+                            'unit': 'year'
+                        },
+                        'count': 1
+                    }
+                ],
+                'organismAgeUnit': [
+                    {
+                        'term': 'year',
+                        'count': 1
+                    }
+                ],
+                'organismAgeValue': [
+                    {
+                        'term': '40-44',
+                        'count': 1
+                    },
+                    {
+                        'term': '55-59',
+                        'count': 1
+                    }
+                ]
+            },
+            # project has 1 donor organism but no organism_age properties
+            'c765e3f9-7cfc-4501-8832-79e5f7abd321': {
+                'organismAge': [
+                    {
+                        'term': {
+                            'value': None,
+                            'unit': None
+                        },
+                        'count': 1
+                    }
+                ],
+                'organismAgeUnit': [
+                    {
+                        'term': None,
+                        'count': 1
+                    }
+                ],
+                'organismAgeValue': [
+                    {
+                        'term': None,
+                        'count': 1
+                    }
+                ],
+            }
+        }
+        self._verify_facet_content(test_data, url)
+
+    def test_organism_age_facet_search(self):
+        """
+        Verify the values of filtered search on organism age facet
+        """
+        url = self.base_url + "/index/projects"
+        test_data = {
+            '627cb0ba-b8a1-405a-b58f-0add82c3d635': {
+                'is': [
+                    {
+                        'value': '20',
+                        'unit': 'year'
+                    }
+                ]
+            },
+            'c765e3f9-7cfc-4501-8832-79e5f7abd321': {
+                'is': [
+                    {
+                        'value': None,
+                        'unit': None
+                    }
+                ]
+            },
+            'invalid_filter_1': {
+                'is': [{}]
+            },
+            'invalid_filter_2': {
+                'is': [
+                    {
+                        'value': None,
+                        'unit': 'weeks'
+                    }
+                ]
+            }
+        }
+        for project_id, filters in test_data.items():
+            with self.subTest(filters=filters):
+                response = requests.get(url, params=dict(catalog=self.catalog,
+                                                         filters=json.dumps({'organismAge': filters})))
+                if project_id.startswith('invalid_filter'):
```

Comment:
Don't use a dict for test_cases but a list of tuples instead. The test cases for invalid data should use `None` for project ID instead of a magic string prefix.